### DISCUSSION TITLE
What is the section argument for in tools.staticdir ?

### DISCUSSION THREAD

**Discussion by davetapley** [cherrypy/cherrypy#2015] [General] [answered]
This requires `section`, but there is no docstring for it:

https://github.com/cherrypy/cherrypy/blob/6387a2b8c3a6413bd88f91f69e1e1a07b8f35def/cherrypy/lib/static.py#L300-L319

---

I found it via the scaffold example, here:
https://github.com/cherrypy/cherrypy/blob/6387a2b8c3a6413bd88f91f69e1e1a07b8f35def/cherrypy/scaffold/__init__.py#L50-L55

---

Playing around with `section` it appears to have no impact on where the route is mounted, the _variable name_ it is assigned to does.

E.g. if we have:

```py
files = tools.staticdir.handler(
        section='/different',
```
then the route is still `/files/whatever.js`, but if I have:

```py
different = tools.staticdir.handler(
        section='/files',
```
then the route is `/different/whatever.js`.

So what is `section` for? 🤔

**Comment by webknjaz** [comment, marked as answer]
I bet it corresponds to the config file section. See https://docs.cherrypy.dev/en/latest/basics.html#serving-a-whole-directory for an INI-style example.

**Reply by davetapley** [reply]
Oh I see, so the part inside the `[]`s?

On closer 👀 its `len` does seem to have some functional impact, via the `branch` variable:
https://github.com/cherrypy/cherrypy/blob/6387a2b8c3a6413bd88f91f69e1e1a07b8f35def/cherrypy/lib/static.py#L346-L350

Then:
https://github.com/cherrypy/cherrypy/blob/6387a2b8c3a6413bd88f91f69e1e1a07b8f35def/cherrypy/lib/static.py#L361

But without docs or tests it's quite hard to figure out the logic there 😑

**Reply by webknjaz** [reply]
The configs aren't necessarily provided via files. You can have that as a key in a dict, if you set things up from the Python land. There's some code mapping those configs to things like plugins too.
In this case, it's about URLs/routes.

**Reply by webknjaz** [reply]
And yes, feel free to submit docs PRs for this. But remember, those Python classes aren't meant to be accessed directly. The end users should just configure the framework and the configs are applied by it under the hood.
